app/uk/gov/hmrc/apidefinition/controllers/DocumentationController.scala

```diff
@@ -30,8 +32,18 @@ class DocumentationController @Inject()(service: DocumentationService, cc: Contr
                                         extends BackendController(cc) {
 
   def fetchApiDocumentationResource(serviceName: String, version: String, resource: String): Action[AnyContent] = Action.async {
-    implicit request =>
+    implicit request => {
       Logger.info(s"API Documentation received request for resource: $serviceName, $version, $resource")
       service.fetchApiDocumentationResource(serviceName, version, resource)
+    } recover recovery
+  }
+
+  private def recovery: PartialFunction[Throwable, Result] = {
+    case e: NotFoundException =>
+      Logger.info(s"Unable to fetch resource: ${e.getMessage}", e)
+      NotFound(error(ErrorCode.API_DEFINITION_NOT_FOUND, e.getMessage))
+    case e =>
```

Comment:
This should probably be allowed to escape as the Play framework will translate this to ISE at the top level.  Also if you retain this is must be case NonFatal(e) =>

Reply:
Have changed this to NonFatal(e)